Smithers.Sessions/JSONHelper.cs

```diff
@@ -107,6 +107,40 @@ public void Serialize(object data, string path)
             }
         }
 
+        public T DeserializeObject<T>(string path) where T : new()
+        {
+            if (!File.Exists(path))
+            {
+                string directory = Path.GetDirectoryName(path);
+
+                if (!Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                    File.Create(path);
+                }
+            }
+
+            try
+            {
+                using (TextReader file = File.OpenText(path))
+                {
+                    JsonSerializer serializer = new JsonSerializer();
+                    T result = (T)serializer.Deserialize(file, typeof(T));
+
+                    if (result == null)
+                    {
+                        result = new T();
+                    }
+
+                    return result;
+                }
+            }
+            catch (IOException)
+            {
+                return new T();
```

Comment:
Failing silently seems like a bad idea. I realize this is copied and pasted from below, but would rather we clean it up.